parser/src/parser.rs

```diff
@@ -394,6 +409,9 @@ pub fn parse_starts_at(
 ///
 /// This could allow you to perform some preprocessing on the tokens before parsing them.
 ///
+/// When in [`Mode::Jupyter`], this will filter out all the Jupyter magic commands
+/// before parsing the tokens.
+///
```

Comment:
Is this temporary or the final solution? I assumed that our plan is to introduce a new MagicCommand ast node.